scalableminds/webknossos
# Add table showing artifacts disk space consumption to workflow view

This PR adds a new table to the vx workflow view that enables users to identify the artifacts consuming the most disk space. The table is wrapped in a modal and can be shown by clicking the "Show Disk Usage of Artifacts" option in workflow view under the menu next to the "Collapse all button" in the header on the right side of the view.


### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- I suggest testing locally, as vx needs to be active on the wk instance.
- Enable voxelytics in the application.conf
- execute ` WK_URL=http://localhost:9000 WK_TOKEN=secretSampleUserToken vx voxelytics/connect/test/configs/segem_test_2d.yaml --cpu` in the working directory of vx and wait for the command to finish
- Go to `http://localhost:9000/workflows` and open the workflow.
- In the "..." menu-button select the option "Show Disk Usage of Artifacts"
- A modal should open up, that displays all artifacts with names, task name, artifact size, inode count, and path in table. Each column of the table should be sortable.
- The table should also work with meta tasks


### Issues:
- fixes [#7069](https://github.com/scalableminds/webknossos/issues/7069)

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Updated [documentation](../blob/master/docs) if applicable <-- is this needed?
